packages/medusa/src/api/routes/admin/users/create-user.js

```diff
@@ -0,0 +1,30 @@
+import { Validator, MedusaError } from "medusa-core-utils"
+import _ from "lodash"
+
+export default async (req, res) => {
+  const schema = Validator.object().keys({
+    email: Validator.string()
+      .email()
+      .required(),
+    name: Validator.string().optional(),
+    password: Validator.string().required(),
+  })
+
+  const { value, error } = schema.validate(req.body)
+  if (error) {
+    throw new MedusaError(MedusaError.Types.INVALID_DATA, error.details)
+  }
+
+  try {
+    const userService = req.scope.resolve("userService")
+    const data = _.pick(value, ["email", "name"])
+
+    let user = await userService.create(data, value.password)
+    user = await userService.retrieve(user._id)
```

Comment:
Is this necessary?

Reply:
The entire `value` object contains password, which is not used to create a user, but needed to hash the password in the service. I thought, this was the neatest way of doing it, but if you have a recommendation, let me know :)

Reply:
Ah got you! But the password isn't returned by `userService.create` right? So I am thinking that we might be able to just remove line 23.

Reply:
Otherwise, I need to make the separation of values in the service instead